src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,135 @@
-        return null;
+        validator(user);
+        return storageDao.add(user);
+    }
+
+    private void validator(User user) {
+        checkIfUserHasNoData(user);
+        checkIfUserAlreadyExist(user);
+        validateLogin(user);
+        validatePassword(user);
+        validateAge(user);
+    }
+
+    private void checkIfUserHasNoData(User user) {
+        if (user == null) {
+            throw new NoValidDataException("User can't be added. "
+                    + "Please set all fields for users");
+        }
+    }
+
+    private void checkIfUserAlreadyExist(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new NoValidDataException("User with this login already exist");
+        }
+    }
+
+    private void validateLogin(User user) {
+        checkIfUserHasNotLogin(user);
+        checkCharactersInUsersLogin(user);
+        checkIfUserHasInvalidLoginLength(user);
+    }
+
+    private void validatePassword(User user) {
+        checkIfUserHasNotPassword(user);
+        checkCharactersInUsersPassword(user);
+        checkIfUserHasInvalidPasswordLength(user);
+    }
+
+    private void validateAge(User user) {
+        checkIfUserDoesNotSetAge(user);
+        checkIfUserHasInvalidAge(user);
+    }
+
+    private void checkIfUserHasInvalidLoginLength(User user) {
+        int loginLength = user.getLogin().length();
+        if (loginLength < VALID_LOGIN_LENGTH_MIN || loginLength > VALID_LOGIN_LENGTH_MAX) {
+            throw new NoValidDataException("Incorrect login length. "
+                    + "Login must contain at least " + VALID_LOGIN_LENGTH_MIN
+                    + "  and less then " + VALID_LOGIN_LENGTH_MAX + " characters");
+        }
+    }
+
+    private void checkCharactersInUsersLogin(User user) {
+        String login = user.getLogin();
+        boolean onlyLetters = true;
+        for (int i = 0; i < login.length(); i++) {
+            char ch = login.charAt(i);
+            if (!Character.isLetter(ch)) {
+                onlyLetters = false;
+                break;
+            }
+        }
+        if (!onlyLetters) {
+            throw new NoValidDataException("Login must contain only characters");
+        }
+    }
+
+    private void checkIfUserHasNotLogin(User user) {
+        if (user.getLogin() == null) {
+            throw new NoValidDataException("You can't add user without login");
+        }
+    }
+
+    private void checkIfUserHasInvalidPasswordLength(User user) {
+        int passwordLength = user.getPassword().length();
+        if (passwordLength < VALID_PASSWORD_LENGTH_MIN
+                || passwordLength > VALID_PASSWORD_LENGTH_MAX) {
+            throw new NoValidDataException("Incorrect password length. "
+                    + "Login must contain at least " + VALID_PASSWORD_LENGTH_MIN
+                    + "  and less then " + VALID_PASSWORD_LENGTH_MAX + " characters");
+        }
+    }
+
+    private void checkCharactersInUsersPassword(User user) {
```

Comment:
Do you have this requirement in task?

Reply:
> Do you have this requirement in task?

I thought it would be nice to validate characters for the user, so I added this check myself